browser/farbling/brave_dark_mode_fingerprint_protection_browsertest.cc

```diff
@@ -0,0 +1,136 @@
+#include "ui/native_theme/test_native_theme.h"
+
+using brave_shields::ControlType;
+
+const char kEmbeddedTestServerDirectory[] = "dark_mode_block";
+
+class BraveDarkModeFingerprintProtection : public InProcessBrowserTest {
+ public:
+  class BraveContentBrowserClientWithWebTheme
+      : public BraveContentBrowserClient {
+   public:
+    explicit BraveContentBrowserClientWithWebTheme(const ui::NativeTheme* theme)
+        : theme_(theme) {}
+
+   protected:
+    const ui::NativeTheme* GetWebTheme() const override { return theme_; }
+
+   private:
+    const ui::NativeTheme* const theme_;
+  };
+
+  void SetUpOnMainThread() override {
+    InProcessBrowserTest::SetUpOnMainThread();
+
+    content_client_.reset(new ChromeContentClient);
+    content::SetContentClient(content_client_.get());
+    content::SetBrowserClientForTesting(
+        new BraveContentBrowserClientWithWebTheme(&test_theme_));
+
+    host_resolver()->AddRule("*", "127.0.0.1");
+    content::SetupCrossSiteRedirector(embedded_test_server());
+
+    brave::RegisterPathProvider();
+    base::FilePath test_data_dir;
+    base::PathService::Get(brave::DIR_TEST_DATA, &test_data_dir);
+    test_data_dir = test_data_dir.AppendASCII(kEmbeddedTestServerDirectory);
+    embedded_test_server()->ServeFilesFromDirectory(test_data_dir);
+
+    ASSERT_TRUE(embedded_test_server()->Start());
+
+    top_level_page_url_ = embedded_test_server()->GetURL("a.com", "/");
+    dark_mode_url_ =
+        embedded_test_server()->GetURL("a.com", "/dark_mode_fingerprint.html");
+  }
+
+  void TearDown() override {
+    browser_content_client_.reset();
+    content_client_.reset();
+  }
+
+  const GURL& dark_mode_url() { return dark_mode_url_; }
+
+  HostContentSettingsMap* content_settings() {
+    return HostContentSettingsMapFactory::GetForProfile(browser()->profile());
+  }
+
+  void AllowFingerprinting() {
+    brave_shields::SetFingerprintingControlType(
+        content_settings(), ControlType::ALLOW, top_level_page_url_);
+  }
+
+  void BlockFingerprinting() {
+    brave_shields::SetFingerprintingControlType(
+        content_settings(), ControlType::BLOCK, top_level_page_url_);
+  }
+
+  void SetFingerprintingDefault() {
+    brave_shields::SetFingerprintingControlType(
+        content_settings(), ControlType::DEFAULT, top_level_page_url_);
+  }
+
+  content::WebContents* contents() {
+    return browser()->tab_strip_model()->GetActiveWebContents();
+  }
+
+  void NavigateToURLUntilLoadStop(const GURL& url) {
+    ui_test_utils::NavigateToURL(browser(), url);
+    ASSERT_TRUE(WaitForLoadStop(contents()));
+  }
+
+ protected:
+  ui::TestNativeTheme test_theme_;
+
+ private:
+  GURL top_level_page_url_;
+  GURL dark_mode_url_;
+  std::unique_ptr<ChromeContentClient> content_client_;
+  std::unique_ptr<BraveContentBrowserClient> browser_content_client_;
+};
+
+IN_PROC_BROWSER_TEST_F(BraveDarkModeFingerprintProtection, DarkModeCheck) {
+  test_theme_.SetDarkMode(true);
```

Comment:
you should also test with `SetDarkMode(false)` to ensure that it isn't affected by this change

Reply:
Done!